resources/aws/asg_stack.go

```diff
@@ -142,16 +167,19 @@ func (s *ASGStack) CreateOrFail() error {
 
 // Update updates the autoscaling group stack in Cloud Formation if one of the
 // updatable parameters has changed.
-func (s *ASGStack) Update() error {
+func (s *ASGStack) Update() (bool, error) {
```

Comment:
Yes, I see the issue. This pattern is used a lot in AWS operator and it's useful for seeing what the operator is doing when developing / debugging cluster issues.

So I continued it with this change. But in this case it does affect the design a lot and is probably unnecessary with good logging. I'll take another look at it.